tests/integration/device_test.sh

```diff
@@ -4,87 +4,261 @@ include './tests/unit/utils.sh'
+    # Convert the RAM value to IEC format
+    expected_output=$(container_exec "$container" "numfmt --from=si --to=iec ${mem_total_kb}K")
+    # Extract RAM value from the output
+    output=$(printf "%s" "$output" | grep "Total RAM" | awk '{print $3}')
+    # Assert that the expected RAM value matches the output RAM value
+    assert_equals_helper "'kw device' RAM check failed for ${distro}" "$LINENO" "$expected_output" "$output"
+  done
+}
 
-  # some fields must be ignored because they surely won't match.
-  filter=(
-    'Root filesystem' 'Size' 'Mounted on'                                            # storage fields
-    'Distribution' 'Distribution base' 'Distribution version' 'Desktop environments' # desktop fields
-  )
+function test_kw_device_local_storage()
+{
+  local distro
+  local container
+  local output_fs
+  local output_size
+  local output_mounted
+  local expected_fs
+  local expected_size
+  local expected_mounted
+
+  for distro in "${!results[@]}"; do
+    container="kw-${distro}"
+
+    # Retrieve expected root filesystem from container
+    expected_fs=$(container_exec "$container" "df -h / | tail -n 1 | awk '{print \$1}'")
+    # Extract actual root filesystem from output
+    output_fs=$(printf "%s" "${results[$distro]}" | awk '/Root filesystem:/ {print $3}')
+    # Assert that expected and actual root filesystem values match
+    assert_equals_helper "'kw device' Root filesystem mismatch for ${distro}" "$LINENO" "$expected_fs" "$output_fs"
+
+    # Retrieve expected root filesystem size from container
+    expected_size=$(container_exec "$container" "df -h / | tail -n 1 | awk '{print \$2}'")
+    # Extract actual root filesystem size from output
+    output_size=$(printf "%s" "${results[$distro]}" | awk '/Size:/ {print $2}')
+    # Assert that expected and actual root filesystem size values match
+    assert_equals_helper "'kw device' Root filesystem size mismatch for ${distro}" "$LINENO" "$expected_size" "$output_size"
+
+    # Retrieve expected root filesystem mounted location from container
+    expected_mounted=$(container_exec "$container" "df -h / | tail -n 1 | awk '{print \$6}'")
+    # Extract actual root filesystem mounted location from output
+    output_mounted=$(printf "%s" "${results[$distro]}" | awk '/Mounted on:/ {print $3}')
+    # Assert that expected and actual root filesystem mounted locations match
+    assert_equals_helper "'kw device' Root filesystem mounted on mismatch for ${distro}" "$LINENO" "$expected_mounted" "$output_mounted"
+  done
+}
+
+function test_kw_device_local_os()
+{
+  local distro
+  local container
+  local expected_distro
+  local output_distro
+
+  for distro in "${!results[@]}"; do
+    container="kw-${distro}"
+
+    # Extract expected distribution name from the container's /etc/os-release
+    expected_distro=$(container_exec "$container" "grep ^NAME= /etc/os-release | cut -d'\"' -f2")
```

Comment:
Please, for this entire commit, review:
https://kworkflow.org/content/codingstyle.html#avoid-short-options